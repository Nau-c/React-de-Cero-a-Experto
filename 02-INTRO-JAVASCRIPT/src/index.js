
const apiKey = '1BqXK4oWRHnjI41los7qLfTUc5dOr9G8' // <-- Add your API key here

const peticion = fetch(`http://api.giphy.com/v1/gifs/random?api_key=${ apiKey }`);

peticion
    .then( resp => resp.json())
    .catch( data => {
        console.log(data);
    })
    .catch(console.warn)

