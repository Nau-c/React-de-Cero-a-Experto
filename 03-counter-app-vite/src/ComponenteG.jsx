import { useState } from 'react'
import PropTypes from 'prop-types';

export const personajes = ['Nau','Iuliia', 'Liam']
export const[ , , p3] = personajes;
console.log(p3)


export const retornaAlgo = () => {
    return ['kk', 123456]

}

export const [letras, numeros] = retronaArreglo();

console.log(letras, numeros)

// Tarea
// 1. el primer valor del arr se llamará nombre
// 2. se llamará setNombre
const useState = ( valor ) => {
    return [ valor, ()=>{ console.log('Hola Mundo') } ];
}

const [ nombre, setNombre ] = useState( 'Goku' );

console.log( nombre );
setNombre();
