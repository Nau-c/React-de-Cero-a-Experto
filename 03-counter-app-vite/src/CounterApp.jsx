import React from "react";
import { useState } from 'react'
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    //hooks son constantes para no setear el valor
    const handleAdd = () => setCounter((count) => count + 1)
    const setRest = () => setCounter((count) => count - 1)
    const setMulti = () => setCounter((count) => count * 2)
    const setDivi = () => setCounter(count => count / 2)

    return (
        <>
            <h1>CounterApp</h1>
            <h2> {counter} </h2>
            <button onClick={handleAdd}> +1 </button>
            <button onClick={setRest}> -1 </button>
            <button onClick={setMulti}> x2 </button>
            <button onClick={setDivi}> /2 </button>
        </>
    );

}