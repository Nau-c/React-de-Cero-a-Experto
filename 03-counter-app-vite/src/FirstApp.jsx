import React from "react";
import PropTypes from "prop-types";

const message = {
    message: 'Probando el objeto',
    title: 'Nauzet'

}

<<<<<<< HEAD
// const getSaludo = () => {
//     if (message) {
//         return <h2>{message.message}</h2>
//     } else {
//         return <h2>Saludo por defecto</h2>
//     }
// }

export const FirstApp = ({ props, title, subTitle, name }) => {
    console.log("PROPS", props);

    if (!title) {
        throw new Error('El titulo es nulo')
=======
const getSaludo = () => {
    if ( message ) {
        return <h2>{ message.message }</h2>
    } else {
        return <h2>Saludo por defecto</h2>
>>>>>>> 7170e88331c3b4d18d4a76a191947602ebbbbffe
    }
}

    export const FirtsApp = ({ title, subTitle, name }) => {
        // console.log( "PROPS", props );
        // subTitle = 'La tierra es grande',
        // title = 'Prueba de titulo'
        // if (!title) {
        //     throw new Error('El title no existe')
        // }

    return (
        <>
            <h1>{ title }</h1>
            {/* <h1>{ getSaludo() }</h1> */}
            {/* <h2>FirtsApp</h2> */}
<<<<<<< HEAD
            <h3>{ subTitle }</h3>
            <p>{ name }</p>
            {/* <p>Soy un subtitulo</p> */}
=======
            {/* <h3>{ title }</h3> */}
            <p>{ subTitle }</p>
            <p>{ name }</p>
>>>>>>> 7170e88331c3b4d18d4a76a191947602ebbbbffe
            {/* <p>{ JSON.stringify( message ) }</p> */}
        </>

    )
}

FirtsApp.propTypes = {
    title: PropTypes.string.isRequired,
<<<<<<< HEAD
    subTitle: PropTypes.number
}

//Los defaultProps son los valores por defecto de las props y entran antes
FirstApp.defaultProps = {
    title: 'Sin titulo',
    subTitle: 'Sin subtitulo',
    name: 'Nauzet'
}
=======
    subTitle: PropTypes.string
}

FirtsApp.defaultProps = {
    name: 'Nauzet',
    subTitle: 'No tiene Subtitulo',
    title: 'No tiene titulo',
}
>>>>>>> 7170e88331c3b4d18d4a76a191947602ebbbbffe
