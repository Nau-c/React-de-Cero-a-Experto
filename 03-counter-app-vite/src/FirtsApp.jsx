import React from "react";

const message = {
    message: 'Probando el objeto',
    title: 'Nauzet'

}

const getSaludo = () => {
    if ( message ) {
        return <h2>{ message.message }</h2>
    } else {
        return <h2>Saludo por defecto</h2>
    }
}

    export const FirtsApp = ( props ) => {
        console.log( props );
    return (
        <>
            <h1>{ getSaludo() }</h1>
            <h2>FirtsApp</h2>
            <p>Soy un subtitulo</p>
            {/* <p>{ JSON.stringify( message ) }</p> */}
        </>

    )
}