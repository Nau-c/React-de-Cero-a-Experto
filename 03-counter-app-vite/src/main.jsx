import React from "react";
import ReactDOM from "react-dom/client";
import { App, HelloWorldApp } from "./HelloWorldApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <App />
    <HelloWorldApp />
  </React.StrictMode>
);
