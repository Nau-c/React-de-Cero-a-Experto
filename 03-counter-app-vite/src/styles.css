html,
body {
    background-color: aliceblue;
    color: rgb(236, 187, 127);
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;

}