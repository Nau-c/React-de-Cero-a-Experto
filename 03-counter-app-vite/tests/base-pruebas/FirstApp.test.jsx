import { getByText, render } from "@testing-library/react";
import { FirstApp } from "../../src/FirstApp";

describe('Pruebas en el archivo FirstApp.test.jsx', () => {

    test('debe de hacer match con el snapshot', () => {

        const title = 'Hola, soy Goku';
        const { container, getByText } = render(<FirstApp title={title} />)
        // console.log(container);

        expect(container).toMatchSnapshot();
    });

    test('debe mostrar el titulo en h1', () => {

        const title = 'Hola, soy Goku';
        const { container, getByText } = render(<FirstApp title={title}/>)
        // console.log(container);
        expect(getByText(title)).toBeTruthy();

        // const h1 = container.querySelector('h1');
        // expect(h1.innerHTML).toContain( title );
        // expect(h1.innerHTML).toBe( title );
        // console.log(h1.innerHTML).toContain( title );
        // console.log(h1.innerHTML).toBe( title );

    });

});