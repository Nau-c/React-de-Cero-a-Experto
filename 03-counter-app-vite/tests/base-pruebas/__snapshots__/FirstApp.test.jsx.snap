// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo FirstApp.test.jsx debe de hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    Hola, soy Goku
  </div>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
